packages/react-server/src/ReactFlightServer.js

```diff
@@ -1334,16 +1336,14 @@ function renderModelDestructive(
             // but that is able to reuse the same task if we're already in one but then that
             // will be a lazy future value rather than guaranteed to exist but maybe that's good.
             const newId = outlineModel(request, (value: any));
-            return serializeByValueID(newId);
+            return request.emittedModelChunkIds.has(newId)
+              ? serializeByValueID(newId)
+              : serializeLazyID(newId);
```

Comment:
I'm unsure about this change. Shouldn't a model that wasn't outlined yet always be referenced by a lazy ID?

This case, as opposed to the case below, is also not covered by the added unit test.

Reply:
Ah, I guess it could be outlined synchronously as part of the `outlineModel` call? So then we never need a lazy ID here, and this line can be reverted?

Reply:
Actually, to fix this in https://github.com/unstubbable/mfng-ai-demo/pull/4 I do not this line to be changed as well. Would be good to add another unit test for this case.

Reply:
Ah, interesting, the change in this line is needed when running in production mode (i.e. `yarn test -r stable --env=production`).